### PR TITLE
feat(profiling): warn user if profiling started before gevent monkey patching

### PR DESCRIPTION
**Pull request by jd**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -179,6 +179,9 @@ def get_exts_for(name):
                 "pyddprofile = ddtrace.profiling.__main__:main",
             ],
             "pytest11": ["ddtrace = ddtrace.contrib.pytest.plugin"],
+            "gevent.plugins.monkey.did_patch_all": [
```
Comment (Kyle-Verhoog) [inline setup.py:182]: Today I learned this existed! We should probably do something similar in the tracer (in a follow up)

File: ddtrace/profiling/profiler.py [hunk 2]
```diff
@@ -55,6 +57,15 @@ def __repr__(self):
 ProfilerStatus.RUNNING = ProfilerStatus("running")
 
 
+def gevent_patch_all(event):
+    if "ddtrace.profiling.auto" in sys.modules:
+        warnings.warn(
+            "Starting the profiler before using gevent monkey patching is not supported "
+            "and is likely to break the application.",
+            RuntimeWarning,
```
Comment (Kyle-Verhoog) [inline ddtrace/profiling/profiler.py:65]: Is it worth mentioning the `DD_GEVENT_PATCH_ALL` feature to help remedy the situation? https://ddtrace.readthedocs.io/en/latest/integrations.html#gevent

**Review by Kyle-Verhoog — COMMENTED** [review]
This is great, didn't know about the gevent hooks!

File: ddtrace/profiling/profiler.py (hunk 2 above)
Comment (jd) [reply ddtrace/profiling/profiler.py:65]: Good idea!

File: releasenotes/notes/profiler-gevent-warns-on-late-patch-3f4f897c8904591c.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+features:
+  - |
+    The `ddtrace.profiling.auto` module will warn users if gevent monkey
+    patching is done after the profiler is a
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/profiler-gevent-warns-on-late-patch-3f4f897c8904591c.yaml:5]: after the profile is a what??? hahah such a cliff hanger 😆 

```suggestion
    patching is done after the profiler is auto-instrumented.
```
?

Comment (jd) [reply releasenotes/notes/profiler-gevent-warns-on-late-patch-3f4f897c8904591c.yaml:5]: lol
the perks of being interrupted all the time 😆

**Review by Kyle-Verhoog — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply setup.py:182]: Also I wonder if we can offer something similar to solve the issue @zhammer had with https://github.com/DataDog/dd-trace-py/issues/1344

**Pull request merged**